sdk/identity/identity/src/util/safeJSONParse.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Safe JSON parse.
+ * @internal
+ */
+export function parse<T>(input: string | null | undefined): T {
```

Comment:
I think we have a safe json parser somewhere that works with recursive structures. @xirzec could you please point to it? we could even consider moving it to core-util.

Reply:
I don't need recursive structures 🤔

Reply:
You probably do not but I would not be so sure, at least to future-proof it. I am fine with what you have for now until we could have a single safe parse method exported somewhere.

Reply:
The recursive problem is with `JSON.stringify()` since that will throw an exception on trying to convert a recursive JS object to a string. The other direction isn't a problem, since JSON itself doesn't support recursion. The worst you can have is invalid JSON.

Reply:
ops, what was I thinking! I will mark this as resolved.